johnpapa/angular-ngrx-data
# Fix (most of) the broken links in the docs

## Purpose

* ...

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Follow the links changed and ensure that they now work. Alternatively use [markdown-link-check](https://www.npmjs.com/package/markdown-link-check).

* Test the code

```
cd docs
find . -name \*.md -exec markdown-link-check {} \;
```

## What to Check
See that there only two broken links relating to `entity-reducer` instead of the previous ~17

```
FILE: ./architecture.md
[✓] entity-metadata.md
[✓] images/action-flow.png
[✓] entity-services.md
[✓] entity-actions.md
[✓] entity-reducer.md
[✓] entity-effects.md
[✓] entity-dataservice.md

FILE: ./entity-actions.md
[✓] entity-services.md
[✓] https://github.com/ngrx/platform/blob/master/docs/store/actions.md
[✓] ../lib/src/actions/entity-action.ts
[✓] entity-metadata.md
[✓] ../lib/src/actions/entity-op.ts
[✓] #action-error
[✓] #action-skip
[✖] ../lib/src/reducers/entity-reducer.ts
[✓] ../lib/src/effects/entity-effects.ts
[✓] #where-are-the-entityactions
[✓] https://github.com/ngrx/platform/tree/master/docs/store-devtools
[✓] https://github.com/ngrx/platform/blob/master/docs/store/actions.md#action-reducers
[✓] https://github.com/ngrx/platform/blob/master/docs/effects/README.md
[✓] https://github.com/ngrx/platform/blob/master/docs/entity/README.md
[✓] https://github.com/ngrx/platform/blob/master/docs/entity/adapter.md#adapter-collection-methods
[✓] entity-change-tracker.md
[✓] entity-reducer.md#entity-cache-actions

ERROR: dead links found!

FILE: ./entity-change-tracker.md
[✓] #enable-change-tracking
[✓] #delete
[✓] #undo

FILE: ./entity-collection-service.md
[✓] ../lib/src/entity-services/entity-collection-service.ts
[✓] entity-actions.md
[✓] ../lib/src/dispatchers/entity-commands.ts
[✓] ../lib/src/selectors/entity-selectors$.ts
[✓] #entity-collection-service-factory
[✓] entity-dataservice.md
[✓] ../lib/src/entity-services/entity-services.ts

FILE: ./entity-collection.md
[✓] ../lib/src/reducers/entity-collection.ts
[✓] entity-change-tracker.md
[✓] entity-metadata.md#additional-collection-state

FILE: ./entity-dataservice.md
[✓] ../lib/src/dataservices/entity-data.service.ts
[✓] ../lib/src/dataservices/entity-data.service.spec.ts
[✓] ../lib/src/dataservices/default-data.service.ts
[✓] #configuration
[✓] ../lib/src/dataservices/http-url-generator.ts
[✓] ../lib/src/utils/default-pluralizer.ts
[✓] entity-metadata.md#plurals

FILE: ./entity-effects.md
[✓] https://github.com/ngrx/platform/blob/master/docs/effects/README.md
[✓] ../lib/src/effects/entity-effects.ts

FILE: ./entity-metadata.md
[✓] #entity-metadata-interface
[✓] entity-actions.md
[✓] entity-dataservice.md
[✓] #plurals
[✓] ../lib/src/entity-metadata/entity-filters.ts
[✖] ../lib/src/reducers/entity-reducer.spec.ts
[✓] entity-collection.md
[✓] ../lib/src/selectors/entity-selectors.spec.ts

ERROR: dead links found!

FILE: ./entity-reducer.md
[✓] #entity-cache-actions
[✓] #initialize
[✓] #collection-reducer-factory
[✓] ../lib/src/reducers/entity-collection-reducer.ts
[✓] entity-metadata.md
[✓] ../lib/src/actions/entity-op.ts
[✓] entity-effects.md
[✓] https://github.com/ngrx/platform/blob/master/docs/entity/adapter.md#adapter-collection-methods
[✓] ../lib/src/reducers/entity-collection-creator.ts
[✓] ../lib/src/entity-metadata/entity-definition.ts
[✓] #register
[✓] #reducer-factory
[✓] https://github.com/ngrx/platform/blob/master/docs/store/api.md#meta-reducers
[✓] #cache-meta-reducer

FILE: ./entity-services.md
[✓] entity-collection-service.md

FILE: ./extension-points.md
[✓] entity-dataservice.md
[✓] entity-reducer.md#customizing
[✓] limitations.md#serialization
[✓] https://angular.io/guide/http#intercepting-requests-and-responses

FILE: ./faq.md
[✓] entity-metadata.md
[✓] #entity
[✓] https://github.com/ngrx/platform/blob/master/README.md
[✓] #redux
[✓] #rxjs
[✓] https://github.com/ngrx/platform/blob/master/docs/store/README.md
[✓] https://github.com/ngrx/platform/blob/master/docs/effects/README.md
[✓] https://github.com/ngrx/platform/blob/master/docs/entity/README.md
[✓] https://redux.js.org/
[✓] #state
[✓] https://en.wikipedia.org/wiki/Immutable_object
[✓] http://reactivex.io/rxjs/

FILE: ./limitations.md
[✓] #alternatives
[✓] https://github.com/ngrx/platform/tree/master/docs/entity
[✓] https://angular.io/guide/http#intercepting-requests-and-responses
[✓] ../lib/src/dispatchers/entity-commands.ts
[✓] http://www.getbreezenow.com/breezejs

FILE: ./publishing-checklist.md
[✓] https://github.com/johnpapa/angular-ngrx-data/releases
```
